sklearn/datasets/samples_generator.py

```diff
@@ -1172,9 +1172,20 @@ def make_sparse_spd_matrix(dim=1, alpha=0.95, norm_diag=False,
         If None, the random number generator is the RandomState instance used
         by `np.random`.
 
+    largest_coef : float between 0 and 1, optional (default=0.9)
+        The value of the largest coefficient.
+
+    smallest_coef : float between 0 and 1, optional (default=0.1)
+        The value of the smallest coefficient.
+
+    norm_diag : boolean, optional (default=False)
+        Whether to normalize the output matrix to make the leading diagonal
+        elements all 1
+
     Returns
     -------
-    prec: array of shape = [dim, dim]
+    prec : sparse matrix of shape = [dim, dim]
```

Comment:
You asked "Is array-like of shape [dim, dim] better?"

Generally, I'd use `of shape (...)` rather than `of shape = (...)`. And while inputs can be array-like, outputs are either arrays or not!